File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        map = defaultdict(int)
        

        if len(s) != len(t):
            return False
        
        for i in range(len(s)):
            map[s[i]] += 1
            map[t[i]] -= 1
        return True if max(map.values()) ==0 else False

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        Set<Character> hashSet = new HashSet<>();
        int longestSubStrLength = 0;

        for(int left = 0, right = 0; right < s.length(); right++) {
            Character ch = s.charAt(right);
            
            while(hashSet.contains(ch)) {
                hashSet.remove(s.charAt(left));
                left++;
            }
            hashSet.add(s.charAt(right));
            longestSubStrLength = Math.max(longestSubStrLength, right - left + 1);
        }
        return longestSubStrLength;
    }
}
